consensus/src/epoch_manager.rs

```diff
@@ -1014,17 +1064,33 @@ impl<P: OnChainConfigProvider> EpochManager<P> {
             .unwrap_or_else(|_| OnChainRandomnessConfig::default_if_missing());
         let jwk_consensus_config = onchain_jwk_consensus_config
             .unwrap_or_else(|_| OnChainJWKConsensusConfig::default_if_missing());
-        let rand_config = self.try_get_rand_config_for_new_epoch(
+        let rand_configs = self.try_get_rand_config_for_new_epoch(
             &epoch_state,
             &onchain_randomness_config,
             dkg_state,
             &consensus_config,
         );
+
+        let (rand_config, fast_rand_config) = match rand_configs {
+            Ok((rand_config, fast_rand_config)) => (Some(rand_config), fast_rand_config),
+            Err(reason) => {
+                if onchain_randomness_config.randomness_enabled() {
+                    error!(
+                        "Failed to get randomness config for new epoch: {:?}",
+                        reason
+                    );
+                }
+                (None, None)
+            },
+        };
+        if fast_rand_config.is_none() && onchain_randomness_config.fast_randomness_enabled() {
+            error!("Failed to get fast randomness config for new epoch");
```

Comment:
if in epoch e fast path is off, and in epoch e+1 it is on, this error will fire which is undesired?